src/Query/Filter/Field.ts

```diff
@@ -360,7 +360,7 @@ export abstract class Field {
      * @param reverse - false for normal group order, true for reverse group order.
      */
     public createGrouper(reverse: boolean): Grouper {
-        return new Grouper(this.fieldNameSingular(), this.grouper(), reverse);
+        return new Grouper(this.fieldNameSingular(), this.grouper(), reverse, this.comparator());
```

Comment:
I thing this will change (improve) the sort order for urgency grouping.

Which will just require confirmation and then a change the docs of group by urgency please.

Reply:
I checked the docs of urgency [here](https://publish.obsidian.md/tasks/Queries/Grouping#Other+task+properties) and [here](https://publish.obsidian.md/tasks/Advanced/Urgency) but didn't find it related to the changes in this PR (Priority group names...).

Do you mean that now sorting of groups from urgency field can use this code? I agree with that, but I guess this should be a different PR.

Reply:
It’s this text:

> - Currently, the groups run from the lowest urgency to highest.
> - You can reverse this with group by urgency reverse.
> - In a future release, the default group order will become from the highest urgency to lowest.

I am reasonably confident that this PR has done point 3 and the above text should be updated.

Reply:
> didn't find it related to the changes in this PR (Priority group names...).

To be more specific, this PR does several things:

1. change the **naming** of groups for the Priority field
2. change the sort of order of grouping for all fields to **sort by task sorting order** instead of by group name alphabetical order

Reply:
You are right! I wrote a test and tried it before and after all the changes and it does change the behaviour indeed. [Test](https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1966/commits/b7f9f2025237bc83d7e142beafc27d153538e8b8) and [docs](https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1966/commits/67a5baff8c65a545f31891fc07ca2c300ddb53bb).

Now I'm thinking that I should've probably split the PR in 2 - one for refactoring with addition of the comparator to the grouper and the other one to fix the Priority group names. May be a third for the Urgency =)

Reply:
Lmk if you wish to have a separate issue for the reversed urgency order =)